### PR TITLE
[contracts] Upgrade Solidity code to 0.5

### PR DESCRIPTION
**Pull request by snario**
This is blocked on [a bug](https://github.com/EthWorks/Waffle/issues/30) that I think is happening in `ethereum-waffle`.

A nice benefit is that Truffle supports Typescript natively now so we don't need to run `tsc`.

In a future PR I'll take advantage of `abi.decode`.

### CODE REVIEW

**Comment by snario** [comment]
Currently blocked until #366 is merged

```
@counterfactual/contracts: contracts/libs/Transfer.sol
@counterfactual/contracts:   39:23    error    Syntax error: unexpected token t    
```

File: packages/apps/contracts/CommitRevealApp.sol [hunk 1]
```diff
@@ -37,15 +37,15 @@ contract CommitRevealApp {
     bytes32 actionHash;
   }
 
-  function isStateTerminal(AppState state)
+  function isStateTerminal(AppState memory state)
```
Comment (mvanderh) [inline packages/apps/contracts/CommitRevealApp.sol]: Is `memory` required now for params?

Comment (snario) [reply packages/apps/contracts/CommitRevealApp.sol]: Yes

Comment (snario) [reply packages/apps/contracts/CommitRevealApp.sol]: https://solidity.readthedocs.io/en/v0.5.0/050-breaking-changes.html#explicitness-requirements

Comment (mvanderh) [reply packages/apps/contracts/CommitRevealApp.sol]: Cool, I like it

**Pull request merged**